### ISSUE TITLE
LDAP support

### ISSUE THREAD

**Issue by Perflyst** [omarroth/invidious#705]
Add support for auth against LDAP

**Comment by omarroth** [comment]
This seems quite out-of-scope to me. Is there a reason for preferring LDAP in particular? Would you mind providing an example use-case?

**Comment by Perflyst** [comment]
People don't have to create new accounts on each service. One account, one login on all services.
As an admin you have the users central. Other apps like nextcloud, tinytinyrss or mastodon support LDAP too

**Comment by PoGo606** [comment]
Same here, I try to let my users use the same account for all the provided services. LDAP is a great help for that ; at least for the authentication part : same login/pass accross all services.

**Comment by HumanG33k** [comment]
Agree too, several organisations want to migrate to their self-hosted services (nextcloud, email, website) and ldap is a key tech for that and with tools like fusiondirectory it’s "easy" now.

**Comment by PipeItToDevNull** [comment]
My use case is that I am a single user, but I host a lot of stuff and want to have one source for my authentication and credentials.

**Comment by Perflyst** [comment]
still relevant

**Comment by joseph-cheung** [comment]
still relevant!

**Comment by joseph-cheung** [comment]
still relevant.
PLZ support LDAP.

**Comment by unixfox** [comment]
Please stop spamming this issue, if you really need LDAP support then you are free to submit a pull request with the code with the LDAP support.

**Comment by joseph-cheung** [comment]
> Please stop spamming this issue, if you really need LDAP support then you are free to submit a pull request with the code containing the LDAP support.

I apologize, but you are also abusing power. 
If you think this is a spam, it is my fault to communicate with you bureaucracy.

**Comment by unixfox** [comment]
> > Please stop spamming this issue, if you really need LDAP support then you are free to submit a pull request with the code containing the LDAP support.
> 
> I apologize, but you are also abusing power.
> If you think this is a spam, it is my fault to communicate with you bureaucracy.

Apologizes accepted, but I'm just moderating this repository because we want to keep the issues clean from useless comments.

The developer that will implement your feature want to read useful comments, he doesn't want to spend extra time filtering the good comments from the bad comments. The time of open source developers is precious because they aren't getting paid, they are just volunteers.

Please keep in mind that we don't own you anything, we are spending our time on things that matter to us, so maybe no one is interested in LDAP and that's why it takes a long time to have this feature.

No need to reply to comment, you want just add a thumbs up if you agree.

**Comment by yagarea** [comment]
Is there any progress on this issue ?

**Comment by unixfox** [comment]
No, would you be interested to contribute in order to add this feature?

**Comment by SamantazFox** [comment]
Updated to take other authentication methods into account (pun not intended).

**Comment by kkwpsi** [comment]
Still relevant. It would be great if the implementation of this feature was tied to mandatory login. (#446).
Currently you can use security like HTTP auth, but the HTTP auth login credentials are useless for the application itself. [FreshRSS](https://freshrss.github.io/FreshRSS/en/admins/09_AccessControl.html) application has a solution for this: a user logged in via HTTP Auth becomes a user in the application itself.

**Comment by BudgieBird** [comment]
Agreed. If there was a way to implement the way Gitea does, that'd be 🔥

**Comment by WirtsLegs** [comment]
+1 to this, OIDC or even just basic header auth (so we can stick it being something like oath2-proxy) would be a big improvement for private instances with a limited community of users

**Comment by knightglyder** [comment]
Giving this a bump and following to increase visibility to SSO authentication is still desirable.

**Comment by Giggitybyte** [comment]
@unixfox @Fijxu @SamantazFox Also voicing my support for any sort of external authentication method, whether that be LDAP, OAuth, OIDC, or reverse proxy auth. 

Though, given that this issue has been open for _7 years_ and the related OAuth PR has been open for _4 years_, is this even being seriously considered by the Invidious team?

**Comment by unixfox** [comment]
Hello,
After discussing with the team.
We are not going to implement oauth in Invidious.
We lack of real free time in the team to maintain the source code and manage the community. Moreover, the contributor samantaz is not actively contributing to Invidious anymore, and she was one of the leading maintainer regarding this part of the code.
This is why we prefer to focus our time on keeping Invidious working with the constant YouTube changes.
You can however create a fork and apply the code from https://github.com/iv-org/invidious/pull/3164 to run in your Invidious instance. Though you will probably need to keep the code updated with each new Invidious code.

**Comment by VoidScute** [comment]
Humm, the real question is how stable and complexe are that kind of auth protocol. For me if they are mature maintenance will be focus on implementation flaw and language/lib maintenance. For Samantaz (currently I have not check commit logs) but if pr are not merge maybe it not motivate people to help maintaining.
